### PR TITLE
chore(core): convert search service to TS

### PR DESCRIPTION
**Pull request by danielpeach**
@anotherchrisberry or @icfantv or @jrsquared please review.

### CODE REVIEW

File: app/scripts/modules/core/search/search.service.js [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-
-    function search(params) {
-      var defaultParams = {
-        pageSize: defaultPageSize
-      };
-
-      return $http({
-        url: SETTINGS.gateUrl + '/search',
-        params: angular.extend(defaultParams, params),
-        timeout: SETTINGS.pollSchedule * 2 + 5000, // TODO: replace with apiHost call
```
Comment (danielpeach) [inline app/scripts/modules/core/search/search.service.js]: The API service uses the same timeout, so I just used `API` directly instead of `$http` here.

**Comment by jrsquared** [comment]
LGTM

File: app/scripts/modules/core/search/search.service.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  totalMatches?: number;
+}
+
+export class SearchService {
+
+  static get defaultPageSize() { return 500; };
+
+  static get $inject() { return ['$log', 'API']; }
+
+  constructor(private $log: ng.ILogService, private API: Api) { }
```
Comment (icfantv) [inline app/scripts/modules/core/search/search.service.ts]: We've started moving to module imports for `ng.*` types, which, I realize now was never officially communicated.  Sorry about that.  Can you please move them all up to the top?  Thanks.

File: app/scripts/modules/core/securityGroup/securityGroupReader.service.ts [hunk 3]
```diff
@@ -416,15 +417,15 @@ export class SecurityGroupReader {
     });
   }
 
-  public loadSecurityGroupsByApplicationName(applicationName: string): any[] {
+  public loadSecurityGroupsByApplicationName(applicationName: string): ng.IPromise<ISecurityGroup[]> {
```
Comment (icfantv) [inline app/scripts/modules/core/securityGroup/securityGroupReader.service.ts]: Same with this file re `ng.*` types.  Thanks.

File: app/scripts/modules/core/search/search.service.ts (hunk 2 above)
Comment (danielpeach) [inline app/scripts/modules/core/search/search.service.ts]: Sure - sounds good.

**Pull request merged**